Handle Download, Share, Add to Cart for Files having special characte…

sandeep-prasad opened the pull request:
The files having special character in the filename, Download, Share and Add to Cart for the files was not successful.
Code Fix: Encoded the paths of the files and the sent it to request parameter and decoded the file paths in the servlet.

kaushalmall commented:
hi @davidjgonzalez, mind looking at this? I checked and everything seems to be OK.

kaushalmall commented:
@sandeep-prasad you will need to create an issue with proper information and then update the changelog to add the issue number.

davidjgonzalez commented on core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/share/impl/EmailShareServiceImpl.java:
```diff
@@ -135,14 +138,23 @@ public final void share(final SlingHttpServletRequest request, final SlingHttpSe
+        } else if (assetEncodedPaths == null || assetEncodedPaths.length == 0) {
             throw new ShareException("At least one asset is required to share");
         }
+        int arrayLength = assetEncodedPaths.length;
```
This looks rather .. verbose? Can't we do a nice stream() online 141... i think its reasonable to catch encoding exceptions and use the string as-is?

```
final String[] assetPaths = Stream.of(shareParameters.get(ASSET_PATHS, String[]{}))
.map(p -> try { return URLDecoder.decode(p, "UTF-8"); } 
                  catch (UnsupportedEncodingException e)  { log.warn("Could not decode path [ {} ], using as is", p); return p; })
.filter(p -> StringUtils::isBlank)
.toArray();
```
or something?

davidjgonzalez commented on ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/modals/share/share.html:
```diff
@@ -78,7 +78,8 @@ <h3>${properties['errorTitle']}</h3>
 				<sly data-sly-list.asset="${share.assets}">
-					<input type="hidden" name="path" value="${asset.path}" />
+				<sly data-sly-use.encodedPath="${'../encodePath.js' @ url=asset.path}"></sly>
+					<input type="hidden" name="path" value="${encodedPath}" />
```
Should we make a Sling Model (some generic "ComponentUtil" model? .. or we could also make a computed property for the Url? ... that might be cleaner?

sandeep-prasad commented:
@davidjgonzalez : 
Incorporated the review comments.
1. Removed JS Use file and added a Sling model, CompUtil.java.
2. Updated the Java 8 code snippet for Iterating the array.

Thanks, Please review.

sandeep-prasad commented:
@davidjgonzalez , @kaushalmall : Hi can we prioritize the pull request.

davidjgonzalez commented:
Closing - Using #305 which cleans up this PR.
